oncall/state.go

```diff
@@ -124,33 +125,44 @@ func (s *state) CalculateShifts(start, end time.Time) []Shift {
 	isOnCall := make(map[string]*Shift)
 	stillOnCall := make(map[string]bool)
 
-	setOnCall := func(userIDs []string) {
+	setOnCall := func(userIDs []string, shiftType assignment.TargetType) {
 		// reset map
 		for id := range stillOnCall {
 			delete(stillOnCall, id)
 		}
 		now := time.Unix(t.Unix(), 0)
 		for _, id := range userIDs {
 			stillOnCall[id] = true
-			s := isOnCall[id]
-			if s != nil {
+			o := isOnCall[id]
+			if o != nil {
 				continue
 			}
```

Comment:
I only did a rename because it was using `s`, which is the same as the function's `(s State)` variable, so I had to choose a different letter to access the state within that scope